src/cargo/core/shell.rs

```diff
@@ -6,6 +6,30 @@ use termcolor::{self, Color, ColorSpec, StandardStream, WriteColor};
 
 use crate::util::errors::CargoResult;
 
+pub enum TtyWidth {
+    NoTty,
+    Known(usize),
+    Guess(usize),
+}
+
+impl TtyWidth {
+    /// Returns the width used with `-Z terminal-width` for the tty.
+    pub fn terminal_width(&self) -> Option<usize> {
+        match *self {
+            TtyWidth::NoTty => None,
+            TtyWidth::Known(width) | TtyWidth::Guess(width) => Some(width),
```

Comment:
```suggestion
            TtyWidth::NoTty | TtyWidth::Guess(_) => None,
            TtyWidth::Known(width) => Some(width),
```

This is the purpose for distinguishing between the two cases.  Wrapping the diagnostics at 60 does not look very good.

I'm also thinking that a different name for `terminal_width()` might be a good idea, to avoid any potential confusion in the future. Maybe something to emphasize this is only for diagnostics?

Reply:
Ah, I see; fixed, thanks!